src/char/inter.cpp

```diff
@@ -848,89 +849,75 @@ int inter_log(const char* fmt, ...)
-					yaml_invalid_warning("inter_config_readConf: Storage definition with invalid ID field in '" CL_WHITE "%s" CL_RESET "', skipping.\n", node, current_file);
-					continue;
-				}
+	if (!existing && (!node["Name"] || !node["Table"])) {
+		yaml_invalid_warning("inter_server_read_db_sub: Invalid storage definition in '" CL_WHITE "%s" CL_RESET "'.\n", node, source);
+		return false;
+	}
 
-				if( id > UINT8_MAX ){
-					yaml_invalid_warning("inter_config_readConf: Storage definition with invalid ID field in '" CL_WHITE "%s" CL_RESET "', skipping.\n", node, current_file);
-					continue;
-				}
+	if (node["Name"])
+		safestrncpy(storage_table->name, node["Name"].as<std::string>().c_str(), NAME_LENGTH);
+	if (node["Table"])
+		safestrncpy(storage_table->table, node["Table"].as<std::string>().c_str(), DB_NAME_LEN);
+	if (node["Max"]) {
+		try {
+			storage_table->max_num = node["Max"].as<uint16>();
+		}
+		catch (const std::exception&) {
+			yaml_invalid_warning("inter_server_read_db_sub: Storage definition with invalid Max field in '" CL_WHITE "%s" CL_RESET "', skipping.\n", node, source);
+			return false;
+		}
+	} else if (!existing)
+		storage_table->max_num = MAX_STORAGE;
 
-				bool existing = inter_premiumStorage_exists(id);
-				auto storage_table = existing ? interserv_config.storages[id] : std::make_shared<s_storage_table>();
+	if (!existing) {
+		storage_table->id = (uint8)id;
+		interserv_config.storages[id] = storage_table;
+	}
 
-				if (!existing && (!node["Name"] || !node["Table"])) {
-					yaml_invalid_warning("inter_config_readConf: Invalid storage definition in '" CL_WHITE "%s" CL_RESET "'.\n", node, current_file);
-					continue;
-				}
-				
-				if (node["Name"])
-					safestrncpy(storage_table->name, node["Name"].as<std::string>().c_str(), NAME_LENGTH);
-				if(node["Table"])
-					safestrncpy(storage_table->table, node["Table"].as<std::string>().c_str(), DB_NAME_LEN);
-				if (node["Max"]) {
-					try {
-						storage_table->max_num = node["Max"].as<uint16>();
-					}
-					catch (const std::exception&) {
-						yaml_invalid_warning("inter_config_readConf: Storage definition with invalid Max field in '" CL_WHITE "%s" CL_RESET "', skipping.\n", node, current_file);
-						continue;
-					}
-				}
-				else if (!existing)
-					storage_table->max_num = MAX_STORAGE;
+	return true;
+}
 
-				if (!existing) {
-					storage_table->id = (uint8)id;
-					interserv_config.storages[id] = storage_table;
-				}
+/**
+ * Loads storages from the inter-server db.
+ */
+void inter_config_readConf(void) {
+	std::vector<std::string> directories = { "conf/" + interserv_config.cfgFile, "conf/import/" + interserv_config.cfgFile };
+	Database db("INTER_SERVER_DB", 1);
 
-				count++;
-			}
-		}
-		ShowStatus("Done reading '" CL_WHITE "%d" CL_RESET "' storage information in '" CL_WHITE "%s" CL_RESET "'\n", count, current_file.c_str());
-	}
+	if (!db.parse(directories, parse_t(inter_server_read_db_sub)))
+		return;
 }
```

Comment:
The return isn't useless here?

Reply:
Maybe adding a comment in all *_read_db functions then? Like
```
// Space to do checks/updates on the initially loaded database
```